core/src/main/java/rdx/works/core/sargon/AccountsExtensions.kt

```diff
@@ -0,0 +1,148 @@
+package rdx.works.core.sargon
+
+import com.radixdlt.sargon.Account
+import com.radixdlt.sargon.AccountAddress
+import com.radixdlt.sargon.AppearanceId
+import com.radixdlt.sargon.DepositAddressExceptionRule
+import com.radixdlt.sargon.DepositRule
+import com.radixdlt.sargon.DerivationPath
+import com.radixdlt.sargon.DerivationPathScheme
+import com.radixdlt.sargon.DisplayName
+import com.radixdlt.sargon.EntityFlag
+import com.radixdlt.sargon.EntityFlags
+import com.radixdlt.sargon.EntitySecurityState
+import com.radixdlt.sargon.FactorSourceId
+import com.radixdlt.sargon.FactorSourceKind
+import com.radixdlt.sargon.LegacyOlympiaAccountAddress
+import com.radixdlt.sargon.NetworkId
+import com.radixdlt.sargon.OnLedgerSettings
+import com.radixdlt.sargon.PublicKey
+import com.radixdlt.sargon.ResourceAddress
+import com.radixdlt.sargon.ThirdPartyDeposits
+import com.radixdlt.sargon.extensions.HDPathValue
+import com.radixdlt.sargon.extensions.contains
+import com.radixdlt.sargon.extensions.default
+import com.radixdlt.sargon.extensions.getBy
+import com.radixdlt.sargon.extensions.init
+import com.radixdlt.sargon.extensions.nonHardenedIndex
+import com.radixdlt.sargon.extensions.toBabylonAddress
+
+fun Collection<Account>.notHiddenAccounts(): List<Account> = filter { !it.isHidden }
+fun Collection<Account>.hiddenAccounts(): List<Account> = filter { it.isHidden }
+
+val Account.factorSourceId: FactorSourceId
+    get() = securityState.factorSourceId
+
+val Account.derivationPathScheme: DerivationPathScheme
+    get() = securityState.derivationPathScheme
+
+val Account.hasAuthSigning: Boolean
+    get() = securityState.hasAuthSigning
+
+val Account.derivationPathEntityIndex: HDPathValue
+    get() = securityState.transactionSigningFactorInstance.publicKey.derivationPath.nonHardenedIndex
+
+val Account.usesEd25519: Boolean
+    get() = securityState.usesEd25519
+
+val Account.usesSECP256k1: Boolean
+    get() = securityState.usesSECP256k1
+
+val Account.isHidden: Boolean
+    get() = EntityFlag.DELETED_BY_USER in flags
+
+val Account.isOlympia: Boolean
+    get() = usesSECP256k1
+
+val Account.hasAcceptKnownDepositRule: Boolean
+    get() = onLedgerSettings.thirdPartyDeposits.depositRule == DepositRule.ACCEPT_KNOWN
+
+val Account.isLedgerAccount: Boolean
+    get() = securityState.transactionSigningFactorInstance.factorSourceId.kind == FactorSourceKind.LEDGER_HQ_HARDWARE_WALLET
+
+@Suppress("LongParameterList")
+fun Account.Companion.initBabylon(
+    networkId: NetworkId,
+    displayName: DisplayName,
+    publicKey: PublicKey,
```

Comment:
maybe we could use PublicKey.Ed25519 so there will be no mistake what is key curve type?

Reply:
I am not sure if that is 100% true. I will need to double check with Alex for that, but I am pretty sure that we discussed that such temporary method for creating babylon account should use the parent public key class. Will answer on that tomorrow.